plugin/src/main/scala/org/ehcache/integrations/play/EhcacheJCacheWrapperModule.scala

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright Terracotta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.ehcache.integrations.play
+
+import javax.cache.configuration.{Configuration => JCacheConfiguration}
+import javax.inject.{Inject, Provider, Singleton}
+
+import org.ehcache.xml.XmlConfiguration
+import play.api.inject.Module
+import play.api.{Configuration, Environment}
+
+import scala.concurrent.duration.Duration
+
+
+trait EhcacheJCacheWrapperComponents {
+  def environment: Environment
+
+  def configuration: Configuration
+
+  lazy val ehcacheJCacheWrapper: JCacheWrapper = new EhcacheJCacheWrapperProvider(environment, configuration).get
+}
+
+/**
+  * EhcacheWrapperModule
+  */
+@Singleton
+class EhcacheJCacheWrapperModule extends Module {
+  def bindings(environment: Environment, configuration: Configuration) = {
+    Seq(
+      bind[JCacheWrapper].toProvider[EhcacheJCacheWrapperProvider]
+    )
+  }
+}
+
+@Singleton
+class EhcacheJCacheWrapperProvider @Inject()(env: Environment, config: Configuration) extends Provider[JCacheWrapper] {
+  lazy val get: JCacheWrapper = {
+    val resourceName = config.getString("play.cache.jcacheConfig")
+    val xmlConfig = resourceName.map(r => new XmlConfiguration(env.resource(r).get, env.classLoader))
```

Comment:
Are we waiting for those changes to join these before merging?  Or are we merging this now?